### DISCUSSION TITLE
Tree Large Data

### DISCUSSION THREAD

**Discussion by johnferguson4444** [primefaces/community#2221] [PrimeReact]
I see that there is the LazyDoc that shows how to only render on expand. Im trying  to see if there is a better way of dealing with a large tree (30000) that is fully expanded and uses the nodetemplate to render the label, im also using a react class component where the tree is. Im currently seeing to just expand all the nodes it can take up to about 10 seconds and also utilizes quite a bit of memory due to all the elements. I have tried windowing and since it still has to go through the render process before it can start to observe them this didnt seem to help. Also tried to batch the expands but yet again react waits and batches them still all at once. The filtering also has some issues when using large trees. What is the best approach that yall have found to deal with large trees in regards to performance and memory?

**Comment by melloware** [comment]
I think the tree was not intended for such large usage or at least that I have seen you are not the first person trying to do this with large volume of nodes.
